core/src/main/java/io/substrait/dsl/SubstraitBuilder.java

```diff
@@ -313,6 +315,37 @@ private Project project(
     return Project.builder().input(input).expressions(expressions).remap(remap).build();
   }
 
+  public Expand expand(Function<Rel, Iterable<? extends Expand.ExpandField>> fieldsFn, Rel input) {
+    return expand(fieldsFn, Optional.empty(), Optional.empty(), input);
+  }
+
+  public Expand expand(
+      Function<Rel, Iterable<? extends Expand.ExpandField>> fieldsFn,
+      List<String> outputNames,
+      Rel input) {
+    return expand(fieldsFn, Optional.empty(), Optional.of(outputNames), input);
+  }
+
+  public Expand expand(
+      Function<Rel, Iterable<? extends Expand.ExpandField>> fieldsFn,
+      Rel.Remap remap,
+      List<String> outputNames,
+      Rel input) {
+    return expand(fieldsFn, Optional.of(remap), Optional.of(outputNames), input);
+  }
+
+  private Expand expand(
+      Function<Rel, Iterable<? extends Expand.ExpandField>> fieldsFn,
+      Optional<Rel.Remap> remap,
+      Optional<List<String>> outputNames,
+      Rel input) {
+    var fields = fieldsFn.apply(input);
+    var expand = Expand.builder().input(input).fields(fields).remap(remap);
+    outputNames.ifPresent(
+        names -> expand.hint(RelCommon.Hint.newBuilder().addAllOutputNames(names).build()));
+    return expand.build();
+  }
```

Comment:
In my opinion I don't think it makes sense to include an explicit `outputNames` parameter like this in these builders.  We should either make variants that pass a full `hints` object, or not include a `hints` param at all and instead have users utilize the pattern of:

```java
Expand.builder.from(
  expand(...)
).hint(...)
.build()
```

I lean towards not including `hints` at all, as I think setting them is very engine specific and if we have `expand` utilize methods with `hints` params it would make sense for all the other methods to also have variants with `hints` and that blows up the method count.